test/integration/newapp_test.go

```diff
@@ -1711,7 +1712,7 @@ func TestNewAppSourceAuthRequired(t *testing.T) {
 	}
 }
 
-func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) string {
+func setupLocalGitRepo(t *testing.T, passwordProtected bool, requireProxy bool) (string, func()) {
 	// Create test directories
 	testDir, err := ioutil.TempDir("", "gitauth")
```

Comment:
One slight difference is that we have configured all of our tests to clean up at startup, not on teardown. This is to keep as much around as possible after a failed run. The only point at which we can be certain that some artifact will not be useful is once someone runs a new suite of tests. Otherwise, we could be deleting useful artifacts that may help someone debugging a flake or failure.